core/src/main/java/com/es/core/order/OrderServiceImpl.java

```diff
@@ -1,18 +1,91 @@
 package com.es.core.order;
 
-import com.es.core.cart.Cart;
+import com.es.core.cart.CartAccessor;
+import com.es.core.cart.service.CartService;
+import com.es.core.model.exception.OutOfStockException;
 import com.es.core.model.order.Order;
+import com.es.core.model.order.OrderItem;
+import com.es.core.model.order.OrderStatus;
+import com.es.core.model.order.dao.OrderDao;
+import com.es.core.model.phone.dao.PhoneDao;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
+import javax.annotation.Resource;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.es.core.model.phone.util.StringUtil.OUT_OF_STOCK_MESSAGE_WITH_PHONE_ID;
+
 @Service
 public class OrderServiceImpl implements OrderService {
+    @Value("${delivery.price}")
+    private int deliveryPrice;
+    @Resource
+    private CartService cartService;
+    @Resource
+    private PhoneDao phoneDao;
+    @Resource
+    private OrderDao orderDao;
+
     @Override
-    public Order createOrder(Cart cart) {
-        throw new UnsupportedOperationException("TODO");
+    public Order createOrder(CartAccessor cart) {
+        Order order = new Order();
+        Random random = new Random();
+
+        order.setOrderItems(cart.getItems()
+                .stream()
+                .map(item -> new OrderItem(random.nextLong(),
+                        item.getPhone(), order, item.getQuantity()))
+                .collect(Collectors.toList()));
+        order.setSubtotal(cart.getTotalCost());
+        order.setDeliveryPrice(BigDecimal.valueOf(deliveryPrice));
+        order.setTotalPrice(order.getSubtotal().add(order.getDeliveryPrice()));
+        order.setStatus(OrderStatus.NEW);
+
+        return order;
     }
 
     @Override
     public void placeOrder(Order order) throws OutOfStockException {
```

Comment:
There is a lot of DB operations in this method. Did you make sure that DB state will be correct in case of error between DB queries?